src/Catalyst.Core.Modules.Dfs/WebApi/V0/Filter/ApiExceptionFilter.cs

```diff
@@ -0,0 +1,93 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading.Tasks;
+using Catalyst.Core.Modules.Dfs.Controllers.V0.Response;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace Catalyst.Core.Modules.Dfs.WebApi.V0.Filter
+{
+    /// <summary>
+    ///     Handles exceptions thrown by a controller.
+    /// </summary>
+    /// <remarks>
+    ///     Returns a <see cref="ApiError" /> to the caller.
+    /// </remarks>
+    public class ApiExceptionFilter : ExceptionFilterAttribute
+    {
+        /// <inheritdoc />
+        public override void OnException(ExceptionContext context)
+        {
+            var statusCode = 500; // Internal Server Error
+            var message = context.Exception.Message;
+            string[] details = null;
+
+            // Map special exceptions to a status code.
+            if (context.Exception is FormatException)
+            {
+                statusCode = 400; // Bad Request
+            }
+            else if (context.Exception is KeyNotFoundException)
+            {
+                statusCode = 400; // Bad Request
+            }
+            else if (context.Exception is TaskCanceledException)
+            {
+                statusCode = 504; // Gateway Timeout
+                message = "The request took too long to process or was cancelled.";
+            }
+            else if (context.Exception is NotImplementedException)
+            {
+                statusCode = 501; // Not Implemented
+            }
+            else if (context.Exception is TargetInvocationException)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add the missing 'else' clause.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)